src/nunavut/lang/cpp/templates/_composite_type.j2

```diff
@@ -61,10 +61,11 @@ struct {% if composite_type.deprecated -%}
         static constexpr bool IsService      = false;
         static constexpr bool IsRequest      = {{ (composite_type == T.request_type) | string | lower }};
         static constexpr bool IsResponse     = {{ (composite_type == T.response_type) | string | lower }};
-{%- else %}
+{% else %}
         static constexpr bool IsServiceType = false;
-{% endif -%}
-    {%- assert composite_type.extent % 8 == 0 %}
+{% endif %}
+        static constexpr const char* FullNameAndVersion() { return "{{ composite_type }}"; }
```

Comment:
Unfortunately I can't make it as simple as could be in c++17:
`static constexpr char FullNameAndVersion[] = "{{ composite_type }}";`

Reply:
Sonar will catch ODR issue with such approach.